streams/shared/src/main/scala/zio/stream/StreamEffect.scala

```diff
@@ -322,19 +331,22 @@ private[stream] object StreamEffect extends Serializable {
       }
     }
 
-  final def fromInputStream(is: InputStream): StreamEffect[Any, Nothing, Byte] =
-    StreamEffect[Any, Nothing, Byte] {
+  final def fromInputStream(
+    is: InputStream,
+    chunkSize: Int = ZStreamChunk.DefaultChunkSize
+  ): StreamEffect[Any, IOException, Chunk[Byte]] =
+    StreamEffect[Any, IOException, Chunk[Byte]] {
       Managed.effectTotal {
-        def pull(): Byte = {
-          val a = is.read()
-          if (a < 0) {
-            end
-          } else {
-            a.toByte
-          }
+        val buf = Array.ofDim[Byte](chunkSize)
+        def pull(): Chunk[Byte] = {
+          val bytesRead = is.read(buf)
```

Comment:
this needs to be wrapped in try/catch and should catch `IOException` and re-throw it  using `StreamEffect.fail`

Reply:
updated, thanks for help